src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,183 @@
+use std::iter;
+use std::io::{Result, Write};
+
+use env::Env;
+use file_saver::save_to_file;
+use library::{self, Library, ToStripperType};
+use library::Type as LType;
+
+use stripper_interface::{FILE, FILE_COMMENT, MOD_COMMENT, TypeStruct};
+
+const TAB_SIZE : usize = 4;
+
+pub fn generate(env: &Env) {
+    let path =  env.config.target_path.join("doc.cmts");
+    println!("Generating documentation {:?}", path);
+    save_to_file(&path, env.config.make_backup,
+        |w| generate_doc(w, &env.library));
+}
+
+fn generate_doc(w: &mut Write, lib: &Library) -> Result<()> {
+    for i in 0..lib.namespaces.len() {
```

Comment:
This `Namespace` thing corresponds to a single `gir` file (a single `<namespace>` element). So `gtk` is one "namespace". When we run the generator on `gtk` it's unnecessary to generate stuff for `gdk` or `glib` etc.